gangplank/ocp/ocp.go

```diff
@@ -56,6 +63,125 @@ func ocpBuildClient() error {
+		"DISABLE_TLS_VERIFICATION=1",
+		"FORCE_UNPRIVILEGED=1",
+		// Tell the tools where to find the home directory
+		fmt.Sprintf("HOME=%s", cosaSrvDir),
+	)
+
+	registry, registryPath := getPushTagless(apiBuild.Spec.Output.To.Name)
+	pushPath := fmt.Sprintf("%s/%s", registry, registryPath)
+
+	l := log.WithFields(
+		log.Fields{
+			"registry":   registry,
+			"push path":  pushPath,
+			"final push": apiBuild.Spec.Output.To.Name,
+		})
+	l.Info("Pushing to remote registry")
+
+	u, err := user.Current()
+	if err != nil {
+		return fmt.Errorf("unable to determine my username: %v", err)
+	}
+	authPath := filepath.Join("run", "containers", u.Uid, "auth.json")
+	baseEnv = append(baseEnv, fmt.Sprintf("REGISTRY_AUTH_FILE=%s", authPath))
+
+	if token, err := ioutil.ReadFile(serviceAccountTokenFile); err == nil {
+		// Default to using the service account token to login to the registry
+		// This method is the most reliable for OCP3 and while its a hack, it ensures that the
+		// the auth is located where the tooling expects it to be.
+		l.Debug("Using token to access registry")
+		loginCmd := []string{"buildah", "login", "--tls-verify=false", "-u", "serviceaccount", "-p", string(token), registry}
+		cmd := exec.CommandContext(ctx, loginCmd[0], loginCmd[1:]...)
+		cmd.Env = baseEnv
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("failed to login into registry: %v", err)
+		}
+	} else if apiBuild.Spec.Output.PushSecret != nil && apiBuild.Spec.Output.PushSecret.Name != "" {
+		// otherwise locate the build push secret. On OCP3 this is not reliable
+		l.Debug("Using named secret for registry access")
+		ac, ns, err := GetClient(ctx)
+		if err != nil {
+			return fmt.Errorf("unable to fetch push secret")
+		}
+		secret, err := ac.CoreV1().Secrets(ns).Get(apiBuild.Spec.Output.PushSecret.Name, v1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to query for secret %s: %v", apiBuild.Spec.Output.PushSecret.Name, err)
+		}
+		if secret == nil {
+			return fmt.Errorf("secret is empty")
+		}
+
+		// Populate the docker config.
+		for k, v := range secret.Data {
+			if k != ".dockercfg" {
+				continue
+			}
+			log.WithFields(log.Fields{
+				"key":         string(k),
+				"secret name": secret.Name,
+			}).Info("Writing push secret")
+
+			authDir := filepath.Dir(authPath)
+			if err := os.MkdirAll(authDir, 0755); err != nil {
+				return fmt.Errorf("failed to create docker configuration directory")
+			}
+
+			if err := ioutil.WriteFile(authPath, v, 0444); err != nil {
+				return fmt.Errorf("failed writing secret %s:%s", secret.Name, k)
+			}
+		}
+	}
+
+	// pushArgs invokes cosa upload code which creates a named tag
+	pushArgs := []string{
+		"/usr/bin/coreos-assembler", "upload-oscontainer",
+		fmt.Sprintf("--name=%s", pushPath),
+	}
+	// copy the pushed image to the expected tag
+	copyArgs := []string{
+		"skopeo", "copy",
+		"--src-tls-verify=false", "--dest-tls-verify=false",
```

Comment:
Would be good to have these key of the same thing the `DISABLE_TLS_VERIFICATION` flag does.

Reply:
Done.